# HG changeset patch
# Parent 52f690fd0bfc55705bae8daa807491f73d4e240e
Change: Change savegame version
* * *
Codechange: Save the NewGRF versions within a map

diff --git a/src/saveload/newgrf_sl.cpp b/src/saveload/newgrf_sl.cpp
--- a/src/saveload/newgrf_sl.cpp
+++ b/src/saveload/newgrf_sl.cpp
@@ -52,6 +52,7 @@
 	    SLE_STR(GRFConfig, filename,         SLE_STR,    0x40),
 	    SLE_VAR(GRFConfig, ident.grfid,      SLE_UINT32),
 	    SLE_ARR(GRFConfig, ident.md5sum,     SLE_UINT8,  16),
+	SLE_CONDVAR(GRFConfig, version,          SLE_UINT32, 151, SL_MAX_VERSION),
 	    SLE_ARR(GRFConfig, param,            SLE_UINT32, 0x80),
 	    SLE_VAR(GRFConfig, num_params,       SLE_UINT8),
 	SLE_CONDVAR(GRFConfig, palette,          SLE_UINT8,  101, SL_MAX_VERSION),
@@ -78,6 +79,7 @@
 		GRFConfig *c = new GRFConfig();
 		SlObject(c, _grfconfig_desc);
 		if (CheckSavegameVersion(101)) c->SetSuitablePalette();
+//		if (CheckSavegameVersion(PATCH_VERSION)) c->version = 0;
 		AppendToGRFConfigList(&grfconfig, c);
 	}
 }
diff --git a/src/saveload/saveload.cpp b/src/saveload/saveload.cpp
--- a/src/saveload/saveload.cpp
+++ b/src/saveload/saveload.cpp
@@ -213,8 +213,9 @@
  *  148   20659
  *  149   20832
  *  150   20857
+ *  151   THIS_PATCH_VERSION
  */
-extern const uint16 SAVEGAME_VERSION = 150; ///< current savegame version of OpenTTD
+extern const uint16 SAVEGAME_VERSION = 151; ///< current savegame version of OpenTTD
 
 SavegameType _savegame_type; ///< type of savegame we are loading
 
